### PR TITLE
Prepping embed user flow for OC users

### PR DESCRIPTION
**Pull request by nickmvincent**
Adds several features to the embed flow:
- populate templates for embed flow
- select Slack channel by name instead of ID
- minor UI changes (not final: will be updated again based on design suggestion)

### CODE REVIEW

**Comment by amyxzhang** [comment]
Sorry, was I supposed to look at this PR for merging or is it still WIP?

**Comment by nickmvincent** [comment]
Ah, sorry, wasn't clear!

Ready for merge, I think, but assuming the visuals will still change quite a bit.

File: policykit/policyengine/views.py [hunk 1]
```diff
@@ -824,49 +824,177 @@ def document_action_recover(request):
...
+    Get policy by name.
+    """
+    from policyengine.models import Policy
+    name = request.GET.get(key_name)
+    try:
+        return Policy.objects.get(name=name)
+    except Policy.DoesNotExist:
+        return(None)
+
+def embed_populate_templates(request):
```
Comment (amyxzhang) [inline policykit/policyengine/views.py:846]: Yeah, I don't want to be creating one-off policies here. Also the policy engine should have no platform-specific logic. This should be a python script runnable from the command line using manage.py and stored in a platform-specific place.

File: policykit/policyengine/views.py [hunk 2]
```diff
@@ -824,49 +824,177 @@ def document_action_recover(request):
...
+
+        PolicyVariable.objects.create(
+            name="pong_message", label="What to say in response to ping", default_value="pong", is_required=True,
+            prompt="What to say in response to ping", type="string", policy=policy)
+
+    return embed_select_template(request)
+
+
+
+def get_channel_options(community_id):
```
Comment (amyxzhang) [inline policykit/policyengine/views.py:913]: same, there shouldn't be platform-specific logic in the policy engine.

File: policykit/policyengine/views.py [hunk 3]
```diff
@@ -824,49 +824,177 @@ def document_action_recover(request):
...
+
+    slack_community = SlackCommunity.objects.get(community_id=community_id)
+    for channel in slack_community.get_conversations()['channels']:
+        if channel['is_channel']: # get only the "channels" (as opposed to group, im, mpim, private)
+            channel_options.append(
+                {'name': channel['name'], 'channel_id': channel['id']}
+            )
+    return channel_options
+
+def embed_select_template(request):
```
Comment (amyxzhang) [inline policykit/policyengine/views.py:928]: with Leijie's changes, this code may need to be changed. we are planning to move away from "is_template=True" policies in general.

**Comment by amyxzhang** [comment]
Ok I think there will need to be a couple changes. This may also need to wait until after Leijie's changes to how we think about policy templates.

**Pull request closed without merging**